### ISSUE TITLE
[docs] Add recipe deprecation policy

### ISSUE THREAD

**Issue by danimtb** [conan-io/conan-center-index#9902]
Related to https://github.com/conan-io/conan-center-index/issues/9900

AFAIK the deprecation policy is still not added to the documentation although we already have some recipes that went through that process.

**Comment by jgsogo** [comment]
I think this is the FAQ: https://github.com/conan-io/conan-center-index/blob/master/docs/faqs.md#what-is-the-policy-for-removing-older-versions-of-a-package

**Comment by danimtb** [comment]
@jgsogo no, that is a FAQ for the removal of old versions of a reference. Here I am referring to the deprecation of the complete recipe that can happen because of different reasons:
- Library changed name.
- Library project moved to another repo under a different name.
- ...
